libindy/src/commands/anoncreds/prover.rs

```diff
@@ -302,6 +323,59 @@ impl ProverCommandExecutor {
         Ok((cred_req_json, cred_req_metadata_json))
     }
 
+    fn set_credential_attr_tag_policy(&self,
+                                      wallet_handle: WalletHandle,
+                                      cred_def_id: &str,
+                                      catpol: Option<&CredentialAttrTagPolicy>,
+                                      retroactive: bool) -> IndyResult<()> {
+        debug!("set_credential_attr_tag_policy >>> wallet_handle: {:?}, cred_def_id: {:?}, catpol: {:?}, retroactive: {:?}", wallet_handle, cred_def_id, catpol, retroactive);
+
+        let res = match catpol {
+            Some(pol) => {
+                self.wallet_service.upsert_indy_object(wallet_handle, cred_def_id, pol)?;
+            },
+            None => {
+                if self.wallet_service.record_exists::<CredentialAttrTagPolicy>(wallet_handle, cred_def_id)? {
+                    self.wallet_service.delete_indy_record::<CredentialAttrTagPolicy>(wallet_handle, cred_def_id)?;
+                }
+            }
+        };
+
+        // Cascade whether we updated policy or not: could be a retroactive cred attr tags reset to existing policy
+        if retroactive {
+            let query_json = format!(r#"{{"cred_def_id": "{}"}}"#, cred_def_id);
+            let mut credentials_search = self.wallet_service.search_indy_records::<Credential>(wallet_handle, query_json.as_str(), &SearchOptions::id_value())?;
+
+            while let Some(credential_record) = credentials_search.fetch_next_record()? {
+                let (_, credential) = self._get_credential(&credential_record)?;
+                let cred_tags = self.anoncreds_service.prover.build_credential_tags(&credential, &catpol);
+                self.wallet_service.update_record_tags(wallet_handle, self.wallet_service.add_prefix("Credential").as_str(), credential_record.get_id(), &cred_tags)?;
+            }
+        }
+
+        debug!("set_credential_attr_tag_policy <<< res: {:?}", res);
+
+        Ok(res)
+    }
+
+    fn get_credential_attr_tag_policy(&self,
+                                      wallet_handle: WalletHandle,
+                                      cred_def_id: &str) -> IndyResult<String> {
```

Comment:
@sklump how do you think about next option:
* change this function to return `IndyResult<Option<String>>` 
* change API appropriately (allow null pointer in result)
* simplify matches at lines 366 and 403

Reply:
I'm not sure is it better or not, may be you already have considered with option

Reply:
I wrote as above because no other prover command implementations return Indy<Option<...>>, and "null" is a JSON string denoting the absence of such a policy.

Please feel free to change it to how you prefer, but note that the anoncreds API call would also have to change to turn the null option into a JSON "null" string in any case.

Reply:
We (me and Stephen) have discussed pro and cons of different approaches here and have decided to keep current implementation.
there is some drawback now: set API consume null pointer to enable default policy, get API return "null" string in similar case. If it will be inconvenient for users set API can be update to accept "null" string as well